plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -225,13 +228,20 @@ else if (config.getPinGlueClientToCurrentRegion()) {
 
     private static AWSCredentialsProvider getAwsCredentialsProvider(GlueHiveMetastoreConfig config)
     {
+        String sessionIdentifier;
+        try {
+            sessionIdentifier = config.getS3SessionIdentifier().replace("${USER}", UserGroupInformation.getCurrentUser().getUserName());
```

Comment:
We don't use UGI to store the current user information. You should use ConnectorIdentity instead.

Did you mean `replaceAll`?

Why IOException is thrown?

Reply:
This will be the default configuration
hive.s3.session-identifier=trino-session

We can also have the config as below
hive.s3.session-identifier=${USER}
hive.s3.session-identifier=trino-session:${USER}

It is 'replace'. We will be replacing the ${USER} with session user.